solarwinds_apm/otlp_metrics_processor.py

```diff
@@ -0,0 +1,190 @@
+    #      https://swicloud.atlassian.net/browse/NH-65061
+    def has_error(self, span: "ReadableSpan") -> bool:
+        """Calculate if this span instance has_error"""
+        if span.status.status_code == StatusCode.ERROR:
+            return True
+        return False
+
+    # TODO Refactor for both inbound and otlp metrics
+    #      https://swicloud.atlassian.net/browse/NH-65061
+    def get_http_status_code(self, span: "ReadableSpan") -> int:
+        """Calculate HTTP status_code from span or default to UNAVAILABLE"""
+        status_code = span.attributes.get(self._HTTP_STATUS_CODE, None)
+        # Something went wrong in OTel or instrumented service crashed early
+        # if no status_code in attributes of HTTP span
+        if not status_code:
+            # TODO change if refactor
+            status_code = self._HTTP_SPAN_STATUS_UNAVAILABLE
+        return status_code
+
+    # TODO Refactor for both inbound and otlp metrics
+    #      https://swicloud.atlassian.net/browse/NH-65061
+    # Disable pylint for compatibility with Python3.7 else TypeError
+    def calculate_transaction_names(
+        self, span: "ReadableSpan"
+    ) -> Tuple[Any, Any]:  # pylint: disable=deprecated-typing-alias
+        """Get trans_name and url_tran of this span instance."""
+        url_tran = span.attributes.get(self._HTTP_URL, None)
+        http_route = span.attributes.get(self._HTTP_ROUTE, None)
+        trans_name = None
+        custom_trans_name = self.calculate_custom_transaction_name(span)
+
+        if custom_trans_name:
+            trans_name = custom_trans_name
+        elif http_route:
+            trans_name = http_route
+        elif span.name:
+            trans_name = span.name
+        return trans_name, url_tran
+
+    # TODO Refactor for both inbound and otlp metrics
+    #      https://swicloud.atlassian.net/browse/NH-65061
+    def calculate_custom_transaction_name(self, span: "ReadableSpan") -> Any:
+        """Get custom transaction name for trace by trace_id, if any"""
+        trans_name = None
+        trace_span_id = W3CTransformer.trace_and_span_id_from_context(
+            span.context
+        )
+        custom_name = self.apm_txname_manager.get(trace_span_id)
+        if custom_name:
+            trans_name = custom_name
+            # TODO change if refactor
+            # Assume SolarWindsInboundMetricsSpanProcessor is
+            # active and is doing the removal of any
+            # custom txn name from cache if not sampled,
+            # so not doing it here for OTLP
+        return trans_name
+
+    def calculate_span_time(
+        self,
+        start_time: int,
+        end_time: int,
+    ) -> int:
+        """Calculate span time in milliseconds (ms) using start and end time
+        in nanoseconds (ns). OTel span start/end_time are optional."""
+        if not start_time or not end_time:
+            return 0
+        return int((end_time - start_time) // 1e6)
```

Comment:
For the apm-proto metrics we definitely needed to convert to microseconds, but [OTel metrics model](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/data-model.md) seems to support nanoseconds so how about let's not do any conversion and pass in just the start/end delta as-is and see what we get :)

Reply:
If I remove the conversion and record `(end_time - start_time)` then the scale bar gets cranked up to the Millions range! What do you think?

OTLP metrics for response_time ([on NA-02](https://my.na-02.cloud.solarwinds.com/158363224169902080/metrics/details?metrics=%5B%7B%22name%22%3A%22trace.service.response_time%22%2C%22aggregateBy%22%3A%22AVG%22%2C%22searchQuery%22%3A%22%22%2C%22search%22%3A%22service.name%3Aflask-metrics-test%20%22%2C%22groupBy%22%3A%5B%22service.name%22%5D%7D%5D&duration=240&endTime=1698425220)) without conversion:

![otlp_response_time](https://github.com/solarwindscloud/solarwinds-apm-python/assets/96076570/eaf125b7-28fb-42f0-b184-235d0f10861b)

...verses APM-proto inbound metrics for response_time ([on NA-01](https://my.na-01.cloud.solarwinds.com/140638900734749696/metrics/details?metrics=%5B%7B%22name%22%3A%22trace.service.response_time%22%2C%22aggregateBy%22%3A%22AVG%22%2C%22groupBy%22%3A%5B%22service.name%22%5D%2C%22search%22%3A%22service.name%3A%5C%22flask-metrics-test%5C%22%22%2C%22searchQuery%22%3A%22service.name%22%7D%5D&duration=3600)):

![apm-proto_response_time](https://github.com/solarwindscloud/solarwinds-apm-python/assets/96076570/ca8e796c-d276-44c7-a9e0-320cb01ecf78)